### PR TITLE
feat(metrics): add metrics for barrier latency at each stage

### PR DESCRIPTION
**Pull request by xxhZs**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

Add barrer_inflight_latency, barrier_sync_latency, and barrier_wait_commit_latency. We can know how much time these stages take.
And add sync_size_every_epoch, we can know the size of S3 written by each epoch
## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Comment by skyzh** [comment]
Would you please take a screenshot of the metrics? I didn't start it by myself for now, but I have some ideas for how to display this.

We can have multiple panels, e.g. Barrier In-Flight p50, Barrier In-Flight p99.

In each of the panel, we stack `barrer_inflight_latency`, `barrier_sync_latency` and `barrier_wait_commit_latency` together.

https://grafana.com/docs/grafana/next/visualizations/time-series/graph-time-series-stacking/

This might help us better understand what's going on.

**Comment by MrCroxx** [comment]
Would you please add some comments about the new-introduced metrics to explain what duration it is measuring? Or the name looks kind of confusing. 🥰

File: src/storage/src/monitor/state_store_metrics.rs [hunk 1]
```diff
@@ -55,6 +55,7 @@ macro_rules! for_all_metrics {
             write_batch_size: Histogram,
             write_build_l0_sst_duration: Histogram,
             write_build_l0_bytes: GenericCounter<AtomicU64>,
+            write_l0_size_every_epoch: Histogram,
```
Comment (hzxa21) [inline src/storage/src/monitor/state_store_metrics.rs:58]: nits: `write_l0_size_per_epoch`

File: src/stream/src/task/stream_manager.rs [hunk 2]
```diff
@@ -185,17 +194,25 @@ impl LocalStreamManager {
...
-        let rx = {
+        let (rx, timer) = {
             let core = self.core.lock();
             let mut barrier_manager = core.context.lock_barrier_manager();
             barrier_manager.remove_collect_rx(epoch)
         };
         // Wait for all actors finishing this barrier.
-        rx.await.unwrap()
+        let result = rx.expect("no rx for local mode").await.unwrap();
+        timer.expect("no timer for test").observe_duration();
```
Comment (hzxa21) [inline src/stream/src/task/stream_manager.rs:204]: In the current implementation, we start timer for `barrier_inflight_latency` when CN receives inject barrier request and call send_barrier and stop the timer when all actors finishing collecting the barrier. I am confused what `barrier_inflight_latency` measures: it is neither the e2e latency for a barrier to flow through the graph, nor the latency for a barrier to flow through local CN. Looks like it is measuring the latency of the barrier flowing through a partial graph.

**Review by hzxa21 — COMMENTED** [review]
Rest LGTM. I think we can merge this PR after addressing all comments.

**Comment by hzxa21** [comment]
@Mergifyio refresh

**Comment by hzxa21** [comment]
@Mergifyio requeue

**Review by hzxa21 — APPROVED** [review]

**Pull request merged**